include/coap3/coap_event.h

```diff
@@ -56,7 +56,24 @@ typedef enum coap_event_t {
 /**
  * (Q-)BLOCK receive errors
  */
-  COAP_EVENT_PARTIAL_BLOCK     = 0x3001
+  COAP_EVENT_PARTIAL_BLOCK     = 0x3001,
+
+/**
+ * Server session state management events
+ */
+
+ /// Called in the CoAP IO loop if a new *server-side* session is created due
+ /// to an incoming connection.
+ ///
+ /// Note that the session might not be a fully established connection yet,
+ /// it might also refer to e.g. a DTLS session in a handshaking stage.
```

Comment:
The libcoap coding style for comments is to use `/*` ... `*/` (each new line beginning with an *), not `//` for single line comments. Please update and squash the commits into a single commit.